src/platforms/ios/iaphandler.mm

```diff
@@ -309,12 +371,24 @@ - (void)request:(SKRequest*)request didFailWithError:(NSError*)error {
   [numberFormatter setLocale:product.priceLocale];
 
   NSString* price = [numberFormatter stringFromNumber:product.price];
-  m_priceValue = QString::fromNSString(price);
-  logger.log() << "Price:" << m_priceValue;
+  QString priceValue = QString::fromNSString(price);
+
+  logger.log() << "Price:" << priceValue;
   [numberFormatter release];
 
-  emit priceValueChanged();
-  emit productsRegistered();
+  Q_ASSERT(m_pendingProducts.contains(productIdentifier));
+  Product& productData = m_pendingProducts[productIdentifier];
+  productData.m_price = priceValue;
+  productData.m_product = product;
+
+  m_registeredProducts.append(productData);
+  m_pendingProducts.remove(productIdentifier);
+
+  if (m_pendingProducts.isEmpty()) {
+    logger.log() << "All the products has been registered";
```

Comment:
In `unknownProductRegistered` we add `(with failures)`. 
Can we be sure that we did not have any failures in a different product when logging this?

Reply:
We do have logs when the registration fails. And also here a bit. Let's land this. We can improve the logging in a follow-up.